rpc/wrtc_server_stream.go

```diff
@@ -392,10 +392,14 @@ func (s *webrtcServerStream) closeWithSendError(err error) (writeErr error) {
 }
 
 func (s *webrtcServerStream) writeHeaders() error {
+	// Grab the RLock to prevent headers from being set while written.
+	s.mu.RLock()
```

Comment:
Let's save ourselves the `RUnlock` on 398 and instead just grab the lock prior to `metadataToProto(s.header)`.